server/game/core/attack/AttackFlow.ts

```diff
@@ -59,22 +57,33 @@ export class AttackFlow extends BaseStepWithPipeline {
             overwhelmDamageOnly = true;
         }
 
-        let damageEvents: GameEvent[];
+        let damageEvents: GameEvent[] = [];
 
         if (overwhelmDamageOnly) {
             damageEvents = [AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).generateEvent(this.attack.target.controller.base, this.context)];
+            this.context.game.openEventWindow(damageEvents, true);
+        } else if (dealsDamageBeforeDefender) {
```

Comment:
I really like this layout of the if/else cases, feels very easy to read

Reply:
i'm glad you see it that way!